amr-wind/physics/SyntheticTurbulence.cpp

```diff
@@ -506,12 +509,18 @@ void SyntheticTurbulence::initialize()
 void SyntheticTurbulence::update()
 {
     BL_PROFILE("amr-wind::SyntheticTurbulence::update");
+
     // Convert current time to an equivalent length based on the reference
     // velocity to determine the position within the turbulence grid
     const amrex::Real cur_time = m_time.new_time() - m_time_offset;
     const amrex::Real eqiv_len =
         m_wind_profile->reference_velocity() * cur_time;
 
+    // Stop update if the current time is past the duration of the synthetic
+    // turbulence and if the duration is positive
```

Comment:
I thought the approach we wanted was to recycle the input turbulence box after it's finished as implemented here https://github.com/Exawind/amr-wind/blob/main/amr-wind/physics/SyntheticTurbulence.cpp#L266


May be this is not consistent throughout the code?

Reply:
Sorry for the confusing language. I was using the word 'duration' to refer to the value of whatever was set in the parameter `m_duration`, not the amount of convective times of the turbulence available in the turbulence box. If `m_duration` is negative, it would indeed recycle the input turbulence box. I just needed a parameter that would allow me to get an isolated patch of turbulence that would be bounded by an undisturbed mean flow both upstream and downstream. So I chose this parameter to be the duration of the injection of the turbulence.